sdk/cosmos/azure-cosmos/azure/cosmos/_synchronized_request.py

```diff
@@ -149,8 +150,14 @@ def _Request(global_endpoint_manager, request_params, connection_policy, pipelin
         # python 3 compatible: convert data from byte to unicode string
         data = data.decode("utf-8")
 
+    if response.status_code == 404:
+        raise errors.CosmosResourceNotFoundError(message=data, response=response)
+    if response.status_code == 409:
+        raise errors.CosmosResourceExistsError(message=data, response=response)
+    if response.status_code == 412:
```

Comment:
I have to dig through all the code paths hitting this, but a 412 response may be because a resource does/does not exist (e.g. if-match: * for update -> ResourceNotFoundError, if-none-match: * -> ResourceExistsError). How confident are we that we are not getting arbitrary/customer provided if-match headers at this point?

Reply:
Thanks @johanste - @christopheranderson do you know the particulars of when the 412 code is used? (Or if there's API reference docs with details you could point me to)